plugin/evm/vm.go

```diff
@@ -449,22 +449,31 @@ func (vm *VM) Initialize(
 
 	var extDataHashes map[common.Hash]common.Hash
 	// Set the chain config for mainnet/fuji chain IDs
-	switch {
-	case g.Config.ChainID.Cmp(params.AvalancheMainnetChainID) == 0:
+	switch chainCtx.NetworkID {
+	case avalanchegoConstants.MainnetID:
 		config := *params.AvalancheMainnetChainConfig
 		g.Config = &config
 		extDataHashes = mainnetExtDataHashes
-	case g.Config.ChainID.Cmp(params.AvalancheFujiChainID) == 0:
+	case avalanchegoConstants.FujiID:
 		config := *params.AvalancheFujiChainConfig
 		g.Config = &config
 		extDataHashes = fujiExtDataHashes
+	case avalanchegoConstants.LocalID:
+		config := *params.AvalancheLocalChainConfig
+		g.Config = &config
+	default:
+		// TODO: This overrides the chain config in the given genesis (in genesisBytes)
+		// Do we want to do this?
+		// config := params.GetChainConfig(chainCtx.NetworkUpgrades, new(big.Int).Set(g.Config.ChainID))
+		// g.Config = config
```

Comment:
this would break many vm tests as vm tests pass their own upgrade config as part of the genesis and expect them to be activated. We need to either start using context.NetworkUpgrade in vm tests or let it be overridden by genesis config (at least for unit test network etc).

Reply:
There is little to no gain respecting the context upgrades as we already have avalanchego upgrades in genesis chain config. The only possible gain is to stop unmarshaling Avalanche upgrades in Coreth ChainConfig, and let them only be passed from snow context (chain config in genesis json would not matter).